libclamav/regex_suffix.c

```diff
@@ -179,8 +184,25 @@ static void destroy_tree(struct node *n)
     FREE(n);
 }
 
-static uint8_t *parse_char_class(const uint8_t *pat, size_t *pos)
+static uint8_t *parse_char_class(const uint8_t *pat, size_t patSize, size_t *pos)
 {
+
+#ifndef inc_pos
+#define add_pos(posPtr, incVal, posMax)                              \
```

Comment:
And Interesting solution to limit duplicate code.  I'm not a huge fan of macros because they're tough to debug. But it doesn't really make sense to do a static inline function for a  single comparison.  I won't complain too much I guess. 

So my only is I think we should capitalize as `INC_POS` and `ADD_POS` like we do with other macros.